src/main/scala/com/databricks/spark/sql/perf/tpcds/TPCDSTables.scala

```diff
@@ -59,484 +56,554 @@ class TPCDSTables(
   dsdgenDir: String,
   scaleFactor: String,
   useDoubleForDecimal: Boolean = false,
-  useStringForDate: Boolean = false)
-  extends Tables(sqlContext, scaleFactor, useDoubleForDecimal, useStringForDate) {
-  import sqlContext.implicits._
+  useStringForDate: Boolean = false,
+  useStringForCharVarchar: Boolean = true)
+  extends Tables(sqlContext, scaleFactor, useDoubleForDecimal, useStringForDate, useStringForCharVarchar) {
 
   val dataGenerator = new DSDGEN(dsdgenDir)
+
+ /**
+  * Datatype mapping for TPC-DS and Spark SQL, fully matching schemas defined in `tpcds.sql` of the
+  * official tpcds toolkit
+  * see more at:
+  *   http://www.tpc.org/tpc_documents_current_versions/pdf/tpc-ds_v2.9.0.pdf
+  *
+  *    |---------------|---------------|
+  *    |    TPC-DS     |  Spark  SQL   |
+  *    |---------------|---------------|
+  *    |  Identifier   |      INT      |
+  *    |---------------|---------------|
+  *    |    Integer    |      INT      |
+  *    |---------------|---------------|
+  *    | Decimal(d, f) | Decimal(d, f) |
+  *    |---------------|---------------|
+  *    |    Char(N)    |    Char(N)    |
+  *    |---------------|---------------|
+  *    |  Varchar(N)   |  Varchar(N)   |
+  *    |---------------|---------------|
+  *    |     Date      |     Date      |
+  *    |---------------|---------------|
+  */
   val tables = Seq(
     Table("catalog_sales",
       partitionColumns = "cs_sold_date_sk" :: Nil,
-      'cs_sold_date_sk          .int,
-      'cs_sold_time_sk          .int,
-      'cs_ship_date_sk          .int,
-      'cs_bill_customer_sk      .int,
-      'cs_bill_cdemo_sk         .int,
-      'cs_bill_hdemo_sk         .int,
-      'cs_bill_addr_sk          .int,
-      'cs_ship_customer_sk      .int,
-      'cs_ship_cdemo_sk         .int,
-      'cs_ship_hdemo_sk         .int,
-      'cs_ship_addr_sk          .int,
-      'cs_call_center_sk        .int,
-      'cs_catalog_page_sk       .int,
-      'cs_ship_mode_sk          .int,
-      'cs_warehouse_sk          .int,
-      'cs_item_sk               .int,
-      'cs_promo_sk              .int,
-      'cs_order_number          .long,
-      'cs_quantity              .int,
-      'cs_wholesale_cost        .decimal(7,2),
-      'cs_list_price            .decimal(7,2),
-      'cs_sales_price           .decimal(7,2),
-      'cs_ext_discount_amt      .decimal(7,2),
-      'cs_ext_sales_price       .decimal(7,2),
-      'cs_ext_wholesale_cost    .decimal(7,2),
-      'cs_ext_list_price        .decimal(7,2),
-      'cs_ext_tax               .decimal(7,2),
-      'cs_coupon_amt            .decimal(7,2),
-      'cs_ext_ship_cost         .decimal(7,2),
-      'cs_net_paid              .decimal(7,2),
-      'cs_net_paid_inc_tax      .decimal(7,2),
-      'cs_net_paid_inc_ship     .decimal(7,2),
-      'cs_net_paid_inc_ship_tax .decimal(7,2),
-      'cs_net_profit            .decimal(7,2)),
+      """
+        |`cs_sold_date_sk` INT,
```

Comment:
Since `ColumnName` does not have a helper method for char/varchar types, I've replaced schema fields with a string schema.